kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java

```diff
@@ -218,7 +246,9 @@ public void onClosed(WebSocket webSocket, int code, String reason) {
         webSocketClosed.set(true);
         LOGGER.debug("Exec Web Socket: On Close with code:[{}], due to: [{}]", code, reason);
         try {
-            closeExecutor();
+            if (cleanupOnClose.get()) {
```

Comment:
To add: if I remove the conditional check and just naively call `cleanUp()`, I get the behaviour I expect -- the future returned by submitting the `InputStreamPumper` is done as soon as the exec is finished in the pod.